filters/include/pcl/filters/impl/uniform_sampling.hpp

```diff
@@ -101,6 +109,9 @@ pcl::UniformSampling<PointT>::applyFilter (PointCloud &output)
     if (leaf.idx == -1)
     {
       leaf.idx = (*indices_)[cp];
+      if (extract_removed_indices_) {
+        removed_indices_->push_back((*indices_)[cp]]);
```

Comment:
This makes no sense. There's no way at this point for you to know if the index in this leaf will be later on substituted in 124. If it ends up never getting replaced, the index should not be removed.

In line 124, like @taketwo pointed out, is where you're sure the index got replaced and dropped.